examples/training/decision_tree/example_creditcard.ipynb

```diff
@@ -0,0 +1,422 @@
+    "from sklearn.metrics import roc_auc_score as score"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2021-03-01T20:58:12.128893Z",
+     "iopub.status.busy": "2021-03-01T20:58:12.128302Z",
+     "iopub.status.idle": "2021-03-01T20:58:14.585268Z",
+     "shell.execute_reply": "2021-03-01T20:58:14.585718Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Reading binary CreditCardFraud dataset (cache) from disk.\n"
+     ]
+    }
+   ],
+   "source": [
+    "dataset = CreditCardFraud(cache_dir=CACHE_DIR)\n",
+    "X_train, X_test, y_train, y_test = dataset.get_train_test_split()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2021-03-01T20:58:14.592259Z",
+     "iopub.status.busy": "2021-03-01T20:58:14.591363Z",
+     "iopub.status.idle": "2021-03-01T20:58:14.882827Z",
+     "shell.execute_reply": "2021-03-01T20:58:14.883229Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of examples: 213605\n",
+      "Number of features: 28\n",
+      "Number of classes:  2\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Number of examples: %d\" % (X_train.shape[0]))\n",
+    "print(\"Number of features: %d\" % (X_train.shape[1]))\n",
+    "print(\"Number of classes:  %d\" % (len(np.unique(y_train))))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "99.83 % of the training transactions belong to class 0\n",
+      "0.17 % of the training transactions belong to class 1\n"
+     ]
+    }
+   ],
+   "source": [
+    "# the dataset is highly imbalanced\n",
+    "labels, sizes = np.unique(y_train, return_counts=True)\n",
+    "print(\"%.2f %% of the training transactions belong to class 0\" % (sizes[0]*100.0/(sizes[0]+sizes[1])))\n",
+    "print(\"%.2f %% of the training transactions belong to class 1\" % (sizes[1]*100.0/(sizes[0]+sizes[1])))\n",
+    "\n",
+    "from sklearn.utils.class_weight import compute_sample_weight\n",
+    "w_train = compute_sample_weight('balanced', y_train)\n",
+    "w_test = compute_sample_weight('balanced', y_test)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2021-03-01T20:58:14.887460Z",
+     "iopub.status.busy": "2021-03-01T20:58:14.886909Z",
+     "iopub.status.idle": "2021-03-01T21:06:31.452583Z",
+     "shell.execute_reply": "2021-03-01T21:06:31.452960Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Training time (sklearn):   5.85 seconds\n",
```

Comment:
Is this a bit fast? Not sure whether we should have examples where the baseline time is so small.